docs/apis/core/hooks/index.md

```diff
@@ -18,6 +18,17 @@ The most common use case for hooks is to allow customisation of standard plugins
 through hook callbacks in local plugins. For example adding a custom institution password
 policy that applies to all enabled authentication plugins through a new local plugin.
 
+## Hook Policies
+
+New hooks added to Moodle from Moodle 4.4 onwards must meet the following rules:
+
+- When describing a hook which happens before or after, the following modifiers are allowed:
+  - `before`
+  - `after`
+- When a modifier is used, it should be used as a _prefix_
+- Hooks belonging to a core subsystem should be located within that subsystem, and _not_ in the `core` subsystem
```

Comment:
It would be good to improve this by providing several examples. 

For example
- Course-related hooks will be placed within `course/classes/hook`.
- Form-related hooks would be in `lib/form/classes/hook` given the form subsystem lives within `lib/form`
- Hooks from core APIs would be within `lib/classes/hook` (and probably recommended under their API folder for better organisation. For example, `core\navigation`-related hooks would be under `lib/classes/hook/navigation`)  

It would also be good if we could add examples that break the rules.